wrappers/rust/icicle-core/src/ntt/tests.rs

```diff
@@ -1,23 +1,29 @@
 use ark_ff::{FftField, Field as ArkField, One};
 use ark_poly::{EvaluationDomain, GeneralEvaluationDomain};
 use ark_std::{ops::Neg, test_rng, UniformRand};
-use icicle_cuda_runtime::device_context::get_default_device_context;
+use icicle_cuda_runtime::device::get_device_count;
+use icicle_cuda_runtime::device::set_device;
 use icicle_cuda_runtime::memory::HostOrDeviceSlice;
-use icicle_cuda_runtime::stream::CudaStream;
+use rayon::iter::IntoParallelIterator;
+use rayon::iter::ParallelIterator;
+
+pub use icicle_cuda_runtime::device_context::get_default_device_context;
 
 use crate::{
-    ntt::{get_default_ntt_config, initialize_domain, ntt, NTTDir, Ordering},
+    ntt::{get_default_ntt_config, get_default_ntt_config_for_device, initialize_domain, ntt, NTTDir, Ordering},
     traits::{ArkConvertible, FieldImpl, GenerateRandom},
 };
 
+use crate::ntt::get_default_context_for_device;
+
 use super::NTT;
 
-pub fn init_domain<F: FieldImpl + ArkConvertible>(max_size: u64)
+pub fn init_domain<F: FieldImpl + ArkConvertible>(max_size: u64, device_id: usize)
```

Comment:
consider adding default value for device_id to avoid passing 0 if having 1 GPU

Reply:
Rust doesn't support default values if I understand you correctly